tests/module.py

```diff
@@ -150,6 +150,23 @@ def my_func(x: JaxArray, y: JaxArray, *, named: int = 1) -> Tuple[JaxArray, JaxA
         self.assertEqual(u.tolist(), [[-2.0, -1.0], [0.0, 1.0], [2.0, 3.0]])
         self.assertEqual(v.tolist(), [[1.0, 0.0], [-1.0, -2.0], [-3.0, -4.0]])
 
+    def test_auto_vars_decorator(self):
+        m = ComplexModule(2)
+
+        @objax.Function.auto_vars
+        def my_func(x: JaxArray, y: JaxArray, *, named: int = 1) -> Tuple[JaxArray, JaxArray]:
+            return m(x, training=True) - named, m(y, training=False) + named
+
```

Comment:
It might be good to add a bug to add more thorough tests for this functionality. It looks good to me and we can commit for now, but having tests to make sure it's actually right would seem helpful.

Reply:
I created a new issue https://github.com/google/objax/issues/206 to track improvements for tracing (including more tests).
If you have specific ideas for test please add them to this issue.